File: Escola/Cursos.Api/scripts/service/eventoService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 1:58:38 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("EventoService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Evento/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Evento/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Evento, callback) {
            $http.post(serviceBase + "Evento/post", Evento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Evento, callback) {
            $http.post(serviceBase + "Evento/put", Evento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Evento, callback) {
            $http.post(serviceBase + "Evento/delete", Evento)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

File: Escola/Cursos.Repository/Services/EventoService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  EventoService
      /// </summary>
      public partial class EventoService : BaseServices, IEventoService
      {

		public EventoService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_EVENTO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_EVENTO> entity)
        {
            UnitOfWork.EventoRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_EVENTO
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_EVENTO entity)
        {	
            UnitOfWork.EventoRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_EVENTO
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_EVENTO entity)
        {
            UnitOfWork.EventoRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_EVENTO
        /// </summary>
        /// <returns>IEnumerable EV_EVENTO</returns>
        public IEnumerable<EV_EVENTO> GetAll()
        {
			return UnitOfWork.EventoRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_EVENTO
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_EVENTO entity)
        {
            UnitOfWork.EventoRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_EVENTO By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_EVENTO</returns>
		public EV_EVENTO GetById(int id)
        {
			return UnitOfWork.EventoRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_EVENTO> FindBy(Expression<Func<EV_EVENTO, bool>> predicate)
        {
            return UnitOfWork.EventoRepository.FindBy(predicate);
        }
      }
}
